extensions/amp-ad-network-adsense-impl/0.1/amp-ad-network-adsense-impl.js

```diff
@@ -165,6 +187,40 @@ export class AmpAdNetworkAdsenseImpl extends AmpA4A {
       addExperimentIdToElement(verifierEid, this.element);
     }
 
+    const correlator = getCorrelator(this.win);
+    this.ampAnalyticsPageLoadMetricsConfig_ = dict({
```

Comment:
By putting this in `buildCallback`, doesn't this enable the fvt ping for _all_ AdSense ad slots/pages?  We want to be able to rate throttle these and control them by experiment flag.  For the current CSI pings, that's handled by the `initLifecycleReporter` call that the constructor and `unlayoutCallback` make, which delegates to `googleLifecycleReporterFactory`, which, in turn, calls `getLifecycleReporter`.  That last does the experiment selection and decides whether or not to enable CSI reporting.  You should tap into that -- either by building/enabling the config directly in `getLifecycleReporter` or by using the experiment-controlled `isReportingEnabled` test that it uses.  (Former is probably better than the latter, because `getLifecyleReporter` has some additional logic that you'd probably want to replicate.)  That way, you guarantee that your reporting is enabled in the same cases that current CSI pings are.  (Though, as above, keep in mind that this is intended to be page-level, rather than slot-level, so regardless of where you do the experiment selection, you should apply the results only once per page.)

Reply:
(Note that my commit earlier this afternoon has not addressed this comment yet - tackled low-hanging fruit first, in anticipation of an interrupt coming soon.)

Reply:
SGTM